tools/raid_xmltolua.cpp

```diff
@@ -0,0 +1,479 @@
+		} catch (const std::runtime_error& err) {
+			std::cout << "[Error - loadRaid] " << info.path << ": " << err.what() << std::endl;
+		}
+	}
+
+	std::sort(eventList.begin(), eventList.end(), [](
+			const std::unique_ptr<RaidEvent>& lhs,
+			const std::unique_ptr<RaidEvent>& rhs
+	) {
+		return lhs->delay < rhs->delay;
+	});
+
+	return {std::move(eventList), std::move(info)};
+}
+
+std::vector<Raid> loadRaids(const fs::path& raids_path)
+{
+	fs::path xml_path = raids_path / "raids.xml";
+	std::cout << xml_path << std::endl;
+
+	pugi::xml_document doc;
+	pugi::xml_parse_result result = doc.load_file(xml_path.c_str());
+	if (not result) {
+		std::cout << "[Error - loadRaids] " << result.description() << std::endl;
+		return {};
+	}
+
+	std::vector<Raid> raidList;
+	for (auto&& raidNode: doc.child("raids").children()) {
+		pugi::xml_attribute attr;
+
+		std::string name;
+		if ((attr = raidNode.attribute("name"))) {
+			name = attr.as_string();
+		} else {
+			std::cout << "[Error - loadRaids] Name tag missing for raid" << std::endl;
+			continue;
+		}
+
+		std::string file;
+		if ((attr = raidNode.attribute("file"))) {
+			file = attr.as_string();
+		} else {
+			std::ostringstream ss;
+			ss << "raids/" << name << ".xml";
+			file = ss.str();
+			std::cout << "[Warning - loadRaids] File tag missing for raid " << name << ". Using default: " << file << std::endl;
+		}
+
+		uint32_t interval = pugi::cast<uint32_t>(raidNode.attribute("interval2").value()) * 60;
+		if (interval == 0) {
+			std::cout << "[Error - loadRaids] interval2 tag missing or zero (would divide by 0) for raid: " << name << std::endl;
+			continue;
+		}
+
+		uint32_t margin = 0;
+		if ((attr = raidNode.attribute("margin"))) {
+			margin = pugi::cast<uint32_t>(attr.value()) * 60 * 1000;
+		} else {
+			std::cout << "[Warning - loadRaids] margin tag missing for raid: " << name << std::endl;
+		}
+
+		bool repeat = false;
+		if ((attr = raidNode.attribute("repeat"))) {
+			repeat = pugi::cast<bool>(attr.as_string());
+		}
+
+		try {
+			Raid&& raid = loadRaid({name, file, interval, margin, repeat}, raids_path);
+			raidList.emplace_back(std::move(raid));
+		} catch (const std::runtime_error& err) {
+			std::cout << "[Error - loadRaids] Failed to load raid \"" << name << "\": " << err.what() << std::endl;
+		}
+	}
+
+	return raidList;
+}
+
+void storeRaids(const fs::path& globalevents_path, const std::vector<Raid>& raidList)
+{
+	for (auto&& raid: raidList) {
+		fs::path script_path = globalevents_path / "scripts" / raid.stem();
+		script_path += ".lua";
+
+		std::cout << script_path << std::endl;
+		std::ofstream output{script_path.native()};
+		if (not output.is_open()) {
+			std::cout << "[Error - storeRaids] Couldn't open file: " << script_path << std::endl;
+			continue;
+		}
+
+		output << raid.to_script() << std::endl;
+	}
+}
+
+int main(int argc, char** argv)
+{
+	fs::path globalevents_path, raids_path;
+
+	int opt;
+	while ((opt = getopt(argc, argv, "hg:r:")) != -1) {
```

Comment:
Since we're already depending on boost, might as well use `boost::program_options` ?